### PR TITLE
Options Menu

### PR DESCRIPTION
**Pull request by AntonAsbrink**
The options menu changes values in the option class and consequently almost all changes require an restart to take effect. I implore anyone to make the changes dynamic if able but otherwise this is what it is.

### CODE REVIEW

File: StortSpelProjekt/Engine/src/Misc/Window.cpp [hunk 1]
```diff
@@ -225,12 +225,12 @@ const HWND* Window::GetHwnd() const
 	return &m_Hwnd;
 }
 
-void Window::SetScreenWidth(int width)
+void Window::SetScreenWidth(const int width)
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Engine/src/Misc/Window.cpp]: Why const? It's not a pointer or a reference, so the source can't be changed anyway..

File: StortSpelProjekt/Game/src/Gamefiles/GameGUI.cpp [hunk 2]
```diff
@@ -33,6 +34,12 @@ void GameGUI::Update(double dt, Scene* scene)
 		}
 	}
 
+	if (scene->GetName() == "OptionScene")
+	{
+		Entity* entity = scene->GetEntity("volume");
+		entity->GetComponent<component::GUI2DComponent>()->GetTextManager()->SetText(Option::GetInstance().GetVariable("f_volume"), "volume");
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Game/src/Gamefiles/GameGUI.cpp:40]: I suppose this is to make the text "volume" appear? It does not show up for me
image

**Review by LeoWikstrom — CHANGES_REQUESTED** [review]
The text "volume" does not show up. It is also possible to set a negative volume, this should be capped to 0, should it not? In my mind it would also be more intuitive to have the + on the right and the - on the left, but that's a personal preference I guess. And finally I assume there was no way to make the green quads move? It looks a bit odd when you select an option and it seems as if nothing happens

File: StortSpelProjekt/Engine/src/Misc/Window.cpp (hunk 1 above)
Comment (AntonAsbrink) [reply StortSpelProjekt/Engine/src/Misc/Window.cpp]: stuff left from trying to make it work dynamically. There was a lot of shenanigans concerning const and windows.

**Review by Whixx — CHANGES_REQUESTED** [review]
Same stuff as Leo said

File: StortSpelProjekt/Game/src/Gamefiles/GameGUI.cpp (hunk 2 above)
Comment (AntonAsbrink) [reply StortSpelProjekt/Game/src/Gamefiles/GameGUI.cpp:40]: There is no text marking that as volume, i was just so used to it being volume i didn't question it

**Review by LeoWikstrom — DISMISSED** [review]
I'd still prefer the + to the right, but everything works. Good job 👍🏽

**Review by Whixx — DISMISSED** [review]
I don't really like that we need this many different textures saved in the project for this to work, but i guess it's fine for now.

**Review by LeoWikstrom — APPROVED** [review]

**Review by Whixx — APPROVED** [review]
brao

**Pull request merged**